synapse/lib/stormlib/cache.py

```diff
@@ -0,0 +1,157 @@
+import asyncio
+
+import synapse.exc as s_exc
+
+import synapse.lib.ast as s_ast
+import synapse.lib.cache as s_cache
+import synapse.lib.stormctrl as s_stormctrl
+import synapse.lib.stormtypes as s_stormtypes
+
+CACHE_SIZE_MAX = 10_000
+CACHE_SIZE_DEFAULT = 10_000
+
+@s_stormtypes.registry.registerLib
+class LibCache(s_stormtypes.Lib):
+    '''
+    A Storm Library for interacting with Cache Objects.
+    '''
+    _storm_locals = (
+        {'name': 'fixed', 'desc': '''
+            Get a new Fixed Cache object.
+
+            When the callback Storm query is executed a special variable,
+            $cache_key, will be set. The query must return a value.
+
+            The fixed cache uses FIFO to evict items once the maximum size is reached.
+
+            Examples:
+
+                // Use a simple query as the callback
+                $cache = $lib.cache.fixed("return(`a value for key={$cache_key}`)")
```

Comment:
If we treat this as the example, then we're providing a example that is syntax checked right away.
```
$cache = $lib.cache.fixed( ${ return(`a value for key={$cache_key}`) } )
```